### PR TITLE
Fix the adjustment of unit answers for comparison.

### PR DESCRIPTION
**Pull request by drgrice1**
Instead of adjusting the correct answer by the correct factor over the student factor, instead adjust the student answer by the the student factor over the correct factor.  Furthermore, only do that temporarily while the comparison is made, and then revert it.

The adjustment is made at a much lower level now.  Instead of doing it in the `cmp_parse` method, do it in the `compare` method which is the overload method for the `<=>` operator.  That is the lowest level of evaluation in the MathObjects answer processing sequence.  This means that it is after the display values for the student answers are set.

Some additional unit tests for this have been added in t/units/basic_parser.t.  Also the "todo" tests that are related to this are now actual tests.

This fixes issue #740.

### CODE REVIEW

**Comment by Alex-Jordan** [comment]
A question based on your description: Suppose the coded correct answer is `1 amu` (which is on the order of 1E-27 kg). Do I understand right that this will be converted to roughly 1E-27 kg before comparison with a student answer (that has also been translated to kg)? If so, would a student answer of `0 amu` or `0 kg` be accepted, with 1E-27 well below the zero tolerance level?

**Comment by Alex-Jordan** [comment]
I haven't done any coding for this issue. But in my head I was outlining an approach where comparison would be made twice. Once using the left operand's units scale, and once using the right operand's units scale. And it would have to pass both tests to count as equal. (You've made me see a third comparison would be natural as well, reducing both operands to the fundamental units scale.) Is there any advantage to an approach like that over what you have done here?

**Comment by drgrice1** [comment]
I switched it to just adjusting the student answer, and not changing the correct at all.  The amu example was problematic.  I am not sure about doing multiple comparisons.  There is a challenge in determining which to trust.

**Comment by Alex-Jordan** [comment]
If we do comparison using the author's unit scale, at least then we can always put responsibility on the author to use a reasonable unit scale given the answer, and/or to adjust context flags like tolType and zeroTolLevel.

**Comment by drgrice1** [comment]
Just to be clear, you mean doing what it is now doing, right?  That is adjusting the student answer to that of the correct (the authors) answer.

**Comment by Alex-Jordan** [comment]
Yes, that's what I mean. Affirming that what you changed it to is going to be good, even if not as thorough as the double comparison I had been considering. The double comparison may not have any real advantage over what you have now.

**Comment by drgrice1** [comment]
After doing some more testing, it seems that the comparison now used by this pull request always gives the correct result.  The others might also be correct, but they are also the ones that may not be correct.  At least in the testing and analysis I have done so far.

**Comment by Alex-Jordan** [comment]
To clarify, when you wrote "correct" just now, do you mean correct behavior from WeBWorK/PG? (There are a few other meanings for "correct" in this thread.)

**Comment by drgrice1** [comment]
Yeah, when I referred to the "correct result" I meant the compare method returns the right thing.  That is 0 if the answers are equal, and nonzero if they aren't.

File: t/units/basic_parser.t [hunk 1]
```diff
@@ -153,34 +150,63 @@ subtest 'Check display methods' => sub {
...
+
+subtest 'Check possible answers compared to the number 0.005 T' => sub {
+	my $correct = NumberWithUnits('0.005 T');
+
+	is check_score($correct, '0.005 T'),        1, '0.005 T is correct';
+	is check_score($correct, '5*10^-13 T*m/A'), 1, '5*10^-13 T*m/A is correct';
+	is check_score($correct, '0 T*m/A'),        0, '0 T*m/A is incorrect';
+};
+
+subtest 'Check possible answers compared to the formula 0.005 T' => sub {
```
Comment (pstaabp) [inline t/units/basic_parser.t:183]: Isn't this subtest exactly the same as the one above?

**Pull request merged**